src/Services/Spotify/spotifyAPI.js

```diff
@@ -1,26 +1,68 @@
 import spotifyConfig from "Config/spotify";
 
-// https://developer.spotify.com/documentation/web-api/reference/#/operations/start-a-users-playback
-function treatHTTPResponse(response) {
-  if (response.status === 200) {
-    return response.json()
-  }
-  throw "API problem"
+const SpotifyWebApi = require('spotify-web-api-node');
+
+const spotifyApi = new SpotifyWebApi({
+  clientId: spotifyConfig.clientId,
+  clientSecret: spotifyConfig.clientSecret,
+  redirectUri: spotifyConfig.redirectUri,
+});
+
+
+function promiseDataHandler(data) {
+  console.log(data);
+  return data.body;
+}
+
+function promiseErrorHandler(err) {
+  console.log(err);
+  throw new Error(`API problem: ${err}`);
```

Comment:
I'm not sure these are necessary at the moment, but maybe they will prove useful in the future. Anyhow, I think they should be used by either all or none of the functions. Currently, `getPlayBackState` doesn't use `promiseDataHandler`, for example (and I think it should)

Reply:
Yeah I put them in here mainly for future use. Worst case we get rid of them along the way.
I'll change ```getPlayBackState```. However we need the extra check with the if ```data.body.isplaying``` otherwise we would return a song that is not currently playing. Or do we not care if the song is currently playing or not?

Reply:
I think the check is maybe easiest done in the component once we write it, which may need some of the data even if it's not playing (is isplaying false when paused? Then the component definitely needs the data to show which song is paused)

Reply:
So in the [SpotifyConsole](https://developer.spotify.com/console/get-user-player/) the first time calling the method with your token it does not return anything but after playing something and pausing it you get two responses where once is_playing is set to true and the other time to false

Reply:
But yes, we can do the check in the component